python/ray/node.py

```diff
@@ -555,6 +563,50 @@ def _prepare_socket_file(self, socket_path, default_prefix):
                               "{} bytes: {!r}".format(maxlen, result))
         return result
 
+    def _get_cached_port(self,
+                         port_name: str,
+                         default_port: Optional[int] = None) -> int:
+        """Get a port number from a cache on this node.  
+    
+        Different driver processes on a node should use the same ports for
+        some purposes, e.g. exporting metrics.  This method returns a port
+        number for the given port name and caches it in a file.  If the
+        port isn't already cached, an unused port is generated and cached.
+
+        Args:
+            port_name (str): the name of the port, e.g. metrics_export_port
+            default_port (Optional[int]): The port to return and cache if no
+            port has already been cached for the given port_name.  If None, an
+            unused port is generated and cached.
+        Returns:
+            port (int): the port number.
+        """
+        file_path = os.path.join(self.get_session_dir_path(),
+                                 "ports_by_node.json")
+
+        # Maps a Node.unique_id to a dict that maps port names to port numbers.
+        ports_by_node: Dict[str, Dict[str, int]] = defaultdict(dict)
```

Comment:
This syntax is not available in python 3.6 I think?

Reply:
My Dev environment is 3.6 so I think it's okay!

Reply:
Hmm not sure what was the issue before, but I definitely remember this wasn't supported... Let's see if CI says something